File: engine/src/main/java/com/terralite/engine/terrain/ChunkLoadRadius.java
package com.terralite.engine.terrain;

public record ChunkLoadRadius(int horizontal, int vertical) {
    public ChunkLoadRadius {
        if (horizontal < 0) {
            throw new IllegalArgumentException("Horizontal chunk load radius cannot be negative");
        }
        if (vertical < 0) {
            throw new IllegalArgumentException("Vertical chunk load radius cannot be negative");
        }
    }

    public static ChunkLoadRadius horizontal(int radius) {
        return new ChunkLoadRadius(radius, 0);
    }

    public static ChunkLoadRadius cubic(int radius) {
        return new ChunkLoadRadius(radius, radius);
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/terrain/ChunkLoadRadius.kt
package com.terralite.engine.terrain

@JvmRecord
data class ChunkLoadRadius(val horizontal: Int, val vertical: Int) {
    init {
        require(horizontal >= 0) { "Horizontal chunk load radius cannot be negative" }
        require(vertical >= 0) { "Vertical chunk load radius cannot be negative" }
    }

    companion object {
        @JvmStatic fun horizontal(radius: Int): ChunkLoadRadius = ChunkLoadRadius(radius, 0)
        @JvmStatic fun cubic(radius: Int): ChunkLoadRadius = ChunkLoadRadius(radius, radius)
    }
}
